src/main/java/uk/gov/hmcts/reform/cmc/model/CmcClaim.java

```diff
@@ -38,11 +47,77 @@ public class CmcClaim {
     private boolean moreTimeRequested;
     private String submitterEmail;
 
+    public Response response;
+    public LocalDate moneyReceivedOn;
+    public LocalDateTime countyCourtJudgmentRequestedAt;
+    public CountyCourtJudgment ccj;
+    public LocalDate admissionPayImmediatelyPastPaymentDate;
+    public ClaimantResponse claimantResponse;
+    public LocalDateTime reDeterminationRequestedAt;
+    public ClaimState state;
+
+    private DefendantResponseStatus status;
+
     public String getClaimantName() {
         return claimData.getClaimantName();
     }
 
     public String getDefendantName() {
         return claimData.getDefendantName();
     }
+
+    public DefendantResponseStatus getDefendantResponseStatus() {
+        if (isEligibleForCCJ()) {
+            return DefendantResponseStatus.ELIGIBLE_FOR_CCJ;
+        }
+        if (hasClaimantRespondedStatesPaid()) {
+            return DefendantResponseStatus.CLAIMANT_ACCEPTED_STATES_PAID;
+        }
+        if (claimantResponse != null && countyCourtJudgmentRequestedAt != null) {
+            return DefendantResponseStatus.REDETERMINATION_BY_JUDGE;
+        }
+        if (moneyReceivedOn != null && countyCourtJudgmentRequestedAt != null && isCCJPaidWithinMonth()) {
+            return DefendantResponseStatus.PAID_IN_FULL_CCJ_CANCELLED;
+        }
+        if (moneyReceivedOn != null && countyCourtJudgmentRequestedAt != null) {
+            return DefendantResponseStatus.PAID_IN_FULL_CCJ_SATISFIED;
+        }
+        if (moneyReceivedOn != null) {
+            return DefendantResponseStatus.PAID_IN_FULL;
+        }
+        if (admissionPayImmediatelyPastPaymentDate != null && claimantResponse == null) {
+            return DefendantResponseStatus.ELIGIBLE_FOR_CCJ_AFTER_FULL_ADMIT_PAY_IMMEDIATELY_PAST_DEADLINE;
+        }
+        if (moreTimeRequested) {
+            return DefendantResponseStatus.MORE_TIME_REQUESTED;
+        }
+        if (state == ClaimState.TRANSFERRED) {
+            return DefendantResponseStatus.TRANSFERRED;
+        }
+        if  (response == null) {
+            return DefendantResponseStatus.NO_RESPONSE;
+        }
+
+        return null;
+    }
+
+    private boolean isEligibleForCCJ() {
+        return ccj != null && ccj.paymentDetails != null;
+    }
+
+    private boolean hasClaimantRespondedStatesPaid() {
+        return claimantResponse != null
+            && claimantResponse == ClaimantResponse.ACCEPTATION
+            && ((response != null
```

Comment:
Extract these sub clauses out into separate methods, e.g. hasClaimaintPaidPartAdmit and hasClaimantPaidFull (or similar)?